README.md

```diff
@@ -73,3 +73,34 @@ To run the integration tests locally:
 - Following the instruction to deploy the `simple-lambda-worker` in the examples readme.
 - Following the instruction to deploy the `simple-authenticated-api` in the examples readme.
 - `npm run integration-test`
+
+## Surviving _commitlint_ & _semantic-release_
+
+_commiitlint_ and _semantic-release_ are two libraries used in the project.
+
+- _commitlint_ currently checks all commit semantics.
+
+- _semantic-release_ will be triggered on commits to main.
+
+### _commitlint_
+
+The commits semantic check is really only relevant when merging (`squash&commit`) PRs into main, but it is currently configured to apply to all commits.
+
+Because the _commitlint_ library can be very restrictive on your commit semantics, for local interin commits (only!), you may want to use the `git commit --no-verify` option.
+
+Remember squashing your local commits and correct the comment so that it follows the _commitlint_ rules before committing & pushing to origin.
+
+For commits squash use `git reset --soft` instead of `git rebase -i`, rebase will most likely fail.
+
+Be aware of GitHub PRs feature _reviewers code suggestions_: accepting & commiting suggestions using the GitHub GUI will generate commits with coauthor information in the comment that will break the _commitlint_ rules.
+
+### _semantic-release_
+
+_Breaking Change_: _semantic-release 17.4.4_ doesn't support _feat!:_ as a _breaking change_ commit type, and it won't recognize it as a _feat_ type either, so as a result it won't publish a new release.
+
+Use the commit comment footer _BREAKING CHANGE:..._ instead. This will trigger a new _Breaking Release_.
+
+To check if your branch follows the _semantic-release_ rules, run the command
+`npx semantic-release -b add_release_label_to_the_namespace --dry-run --branches main`
```

Comment:
The `-b` at the start _is_ the `--branches` flag, but `add_release_label_to_the_namespace` was the name of my branch on talis-cdk8s-constructs... For the check to work you push changes to your branch and then set the flag with your branch name, so that you check before you merge to `main` (which is the default when `--branches` is not set).

:neutral_face:

Reply:
I see! thanks Grz